### ISSUE TITLE
Why is there no field to control the order of containers in the pod of k8s?

### ISSUE THREAD

**Issue by googs1025** [kubernetes/kubernetes#116733]
After searching, k8s can provide the poststart field to control the container field. But if there are more than two containers in the pod and there are order dependencies, the processing will become more complicated. Therefore, would like to ask whether there is a need to provide a field to guarantee the order of pods.

**Comment by charles-chenzz** [comment]
I think statefulset can guarantee the order of pod?

**Comment by googs1025** [comment]
What I want is the order of the container dimension, not the pod dimension

**Comment by kidddddddddddddddddddddd** [comment]
Maybe `init-containers` can help? https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

**Comment by charles-chenzz** [comment]
maybe ``argo`` and ``postStart hook`` can help as well

**Comment by googs1025** [comment]
Yes, I know initContainer with poststart fields can be achieved. But if there are multiple containers, the implementation will be more complicated.

**Comment by kidddddddddddddddddddddd** [comment]
> Yes, I know initContainer with poststart fields can be achieved. But if there are multiple containers, the implementation will be more complicated.

Maybe consider spilt these containers. 
For reasons why kubernetes doesn't handle dependencies of containers, maybe you can find answers in  [#1996](https://github.com/kubernetes/kubernetes/issues/1996), [#1589](https://github.com/kubernetes/kubernetes/issues/1589)

**Comment by BenTheElder** [comment]
See: https://github.com/kubernetes/enhancements/issues/753 and linked issues / PRs for a lengthy history of exploring this space and difficulty settling on a design that is not over complicated and agreeable to all parties involved.

There is very recently some progress but of narrower scope. https://github.com/kubernetes/enhancements/pull/3761, but delayed this release https://github.com/kubernetes/enhancements/issues/753#issuecomment-1469048588

**Comment by googs1025** [comment]
Thank you for your answer. From your answer, I know some historical issues of this topic. In addition, regarding container priority, it is mentioned in this open source project called "openkruise", and you can refer to it if you are interested. https://openkruise.io/docs/v1.2/user-manuals/containerlaunchpriority/
